templates/abandon-resources-hook.yaml

```diff
@@ -25,6 +25,11 @@ spec:
           if kubectl get configmap kotsadm-application-metadata -n {{ .Release.Namespace }} -o jsonpath='{.metadata.labels.app\.kubernetes\.io/managed-by}' | grep -q "Helm"; then
             kubectl annotate configmap kotsadm-application-metadata -n {{ .Release.Namespace }} helm.sh/resource-policy=keep
           fi
+          {{- if .Values.passwordSecretRef }}
+          if kubectl get secret kotsadm-password -n {{ .Release.Namespace }} -o jsonpath='{.metadata.labels.app\.kubernetes\.io/managed-by}' | grep -q "Helm"; then
+            kubectl annotate secret kotsadm-password -n {{ .Release.Namespace }} helm.sh/resource-policy=keep
+          fi
+          {{- end }}
```

Comment:
would this effectively keep the secret around, but it would no longer be used by KOTS? is there a reason why we want to keep it around? kinda thinking out loud here...

Reply:
I didn't think we're deprecating the secret entirely? just pre-generating the secret if needed from EC